File: app/app/src/main/java/com/vassar/unifiedapp/utils/StringUtils.java
package com.vassar.unifiedapp.utils;

import com.vassar.unifiedapp.context.UAAppContext;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class StringUtils {

    public static String getconcatenatedStringFromStringList(String delimitter, List<String> stringList) {
        String concatenatedString = "";
        if (stringList == null || stringList.isEmpty())
            return "";
        else if(stringList.size() == 1)  {
            return stringList.get(0);
        } else {
            for (int i = 0; i < stringList.size()-1; i++) {
                concatenatedString = concatenatedString + stringList.get(i) + delimitter;
            }
            concatenatedString += stringList.get(stringList.size()-1);
        }
        return concatenatedString;
    }
    public static List<String> getStringListFromDelimiter(String delimiter, String concatenatedString) {
        List<String> stringList = new ArrayList<>();
        if (concatenatedString == null || concatenatedString.isEmpty())
            return stringList;
        return Arrays.asList(concatenatedString.split(delimiter));
    }
    public static String getFormattedText(String value) {
        if(value == null)
            return  null;
        value = value.trim();
        value = value.toLowerCase();
        value = value.replace(" ","_");
        return value;
    }
    public static String getTranslatedString(String text) {
        String translatedText = text;
        JSONObject json =  UAAppContext.getInstance().getLocalization();
        if(json != null && json.has(text)) {
            try {
                translatedText = json.getString(text);
            } catch (JSONException e) {
                e.printStackTrace();
            }

        }
        return translatedText;
    }
}

File: uniapp_flutter/lib/utils/string_utils.dart
import '../ua_app_context.dart';

class StringUtils {

  static String getconcatenatedStringFromStringList(String delimitter, List<String> stringList) {
    String concatenatedString = "";
    if (stringList == null || stringList.isEmpty)
      return "";
    else if(stringList.length == 1)  {
      return stringList[0];
    } else {
      for (int i = 0; i < stringList.length-1; i++) {
        concatenatedString = concatenatedString + stringList[i] + delimitter;
      }
      concatenatedString += stringList[stringList.length-1];
    }
    return concatenatedString;
  }
  static List<String> getStringListFromDelimiter(String delimiter, String concatenatedString) {
    List<String> stringList = new List();
    if (concatenatedString == null || concatenatedString.isEmpty)
      return stringList;
    return concatenatedString.split(delimiter);
  }
  static String getFormattedText(String value) {
    if(value == null)
      return  null;
    value = value.trim();
    value = value.toLowerCase();
//    value = value.replace(" ","_");
    return value;
  }
  static String getTranslatedString(String text) {
    String translatedText = text;
    Map<String, dynamic> json =  UAAppContext.getInstance().getLocalization();
    if(json != null && (json.containsKey(text) ||json.containsKey(text.toLowerCase())) ) {
      translatedText = json[text] == null ?json[text.toLowerCase()]: json[text] ;
    }
    return translatedText;
  }

  static List<String> getImageUUIDList(List<String> uuidsWithLongLat) {
    List<String>uuids = List();
    for(String uuidWithLatLong in uuidsWithLongLat) {
      List<String> imageUUIDInfo = getStringListFromDelimiter(
          "##", uuidWithLatLong);
      if (imageUUIDInfo.length > 0) {
        uuids.add(imageUUIDInfo[0]);
      }
    }
    return uuids;
  }
  static String getKey(String key) {
    List<String> subkeys = StringUtils.getStringListFromDelimiter("\$\$", key);
    return subkeys.elementAt(subkeys.length - 1);
  }
}
